### PR TITLE
test: add tests for args.go

### PR DESCRIPTION
**Pull request by rakshitgondwal**
Adds additional test cases for `args.go` in `args_test.go`.

### CODE REVIEW

**Comment by rakshitgondwal** [comment]
@rjsadow Can you please have a look why the tests are failing, cant figure it out.

**Comment by rjsadow** [comment]
Understanding where/how this line came to be: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_hydrophone/140/pull-hydrophone-simple-run/1751961702041128960#1:build-log.txt%3A139. If you also look at line 144, at the end there's a  `,[]` at the end. That shouldn't be there.

**Comment by rjsadow** [comment]
I'm guessing we're not handling a case properly when extra-args are empty.

File: pkg/common/args_test.go [hunk 1]
```diff
@@ -44,7 +47,9 @@ func TestValidateArgs(t *testing.T) {
 			skip:          "",
 			expectedSkip:  "",
 			extraArgs:     []string{},
-			expectedArgs:  "",
+			expectedArgs:  []string{},
```
Comment (reetasingh) [inline pkg/common/args_test.go:50]: @rakshitgondwal expectedArgs is a  a string and not slice of string since that is what the container expects.

**Comment by rakshitgondwal** [comment]
> when running from main branch did you run make build again?

Yep.
Screenshot from 2024-01-30 02-04-30

> the issue would be mostly coming from https://github.com/kubernetes-sigs/hydrophone/blob/main/pkg/common/args.go#L62 if extra-args is not of type []string but of type []interafce{}

This check should not be running in this case right? as I have not defined any `extraArgs`.

**Comment by reetasingh** [comment]
> > when running from main branch did you run make build again?
> 
> Yep. Screenshot from 2024-01-30 02-04-30
> 
> > the issue would be mostly coming from https://github.com/kubernetes-sigs/hydrophone/blob/main/pkg/common/args.go#L62 if extra-args is not of type []string but of type []interafce{}
> 
> This check should not be running in this case right? as I have not defined any `extraArgs`.
> 
> If there is no `extraArgs` defined, the value of that flag inside config is `[]`, and in the logic we are checking `extraArgs != ""`, maybe this is what is causing the error?

this will help https://github.com/kubernetes-sigs/hydrophone/pull/141

**Comment by reetasingh** [comment]
> /hold wait for #141 to be merged first

@rakshitgondwal PTAL https://github.com/kubernetes-sigs/hydrophone/pull/141 , added a cleaner solution. Thanks for highlighting the problem

**Pull request merged**